machines-endpoints/pkg/machines-endpoints/main.go

```diff
@@ -418,4 +422,74 @@ func main() {
 			log.ErrorExit(err)
 		}
 	}
+
+	if *flgBMCLogCollectorConfigMap {
+		// create bmc-log-collector configmap of all servers that include boot servers
+		addersses, err := updateServerAddressList(machines)
+		if err != nil {
+			log.ErrorExit(err)
+		}
+		err = client.writeConfigMap(ctx, addersses)
+		if err != nil {
+			log.ErrorExit(err)
+		}
+	}
+}
+
+func updateServerAddressList(machines []Machine) (map[string]string, error) {
+	type MachineConfigMap struct {
+		Serial string `json:"serial"`
+		BmcIP  string `json:"bmc_ip"`
+		NodeIP string `json:"ipv4"`
+		Role   string `json:"role"`
+		State  string `json:"state"`
+	}
+	var ml []MachineConfigMap
+
+	for _, machine := range machines {
+		var m MachineConfigMap
+		if machine.Spec.BMC.IPv4 == "" {
+			continue
+		}
+		m.Serial = machine.Spec.Serial
+		m.BmcIP = machine.Spec.BMC.IPv4
+		m.NodeIP = machine.Spec.IPv4[0]
+		m.Role = machine.Spec.Role
+		m.State = machine.Status.State
+		ml = append(ml, m)
+	}
+
+	byteJSON, err := json.Marshal(ml)
+	if err != nil {
+		return nil, err
+	}
+	addresses := make(map[string]string)
+	addresses["serverlist.json"] = string(byteJSON)
+	return addresses, nil
+}
```

Comment:
IMO, it is better to return the contents of `server-list JSON`.

It is not good to depend on the structure of the ConfigMap.
Wrapping with a map for ConfigMap should be done in `writeConfigMap()`.

```suggestion
	byteJSON, err := json.Marshal(ml)
	if err != nil {
		return nil, err
	}
	string(byteJSON), nil
}
```